client/src/modules/dashboard/Events/components/EventCancelButton.tsx

```diff
@@ -13,7 +13,30 @@ interface EventCancelButtonProps extends ButtonProps {
 const EventCancelButton: React.FC<EventCancelButtonProps> = (props) => {
   const { isFullWidth, buttonText, event, ...rest } = props;
 
-  const [cancel] = useCancelEventMutation();
+  const [cancel] = useCancelEventMutation({
+    update(cache, { data: canceledEvent }) {
+      cache.modify({
+        fields: {
+          events(existingData) {
+            cache.writeFragment({
+              id: `EventWithEverything:${canceledEvent?.cancelEvent.id}`,
+              data: canceledEvent?.cancelEvent,
+              fragment: EVENT_CANCEL_FRAGMENT,
+            });
+            return existingData;
+          },
+          paginatedEvents(existingData) {
+            cache.writeFragment({
+              id: `EventWithChapter:${canceledEvent?.cancelEvent.id}`,
+              data: canceledEvent?.cancelEvent,
+              fragment: EVENT_CANCEL_FRAGMENT,
+            });
+            return existingData;
```

Comment:
I'm gonna write here to keep discussion complete, but example will be based on the case with editing event.

From my tests returning data is relevant in some way here.
- Load `http://localhost:3000`, press `F5` to clear cache. Two events on the main page will have `EventWithChapter:${id}` cache.
- Go to `Dashboard` -> `Events` -> one of the two first events, edit event name and submit.
- `EventWithChapter:${id}` key is not updated, what will be visible on the home page.

What I think is happening, without the returned data, is query loading events in the dashboard is run again after edit, what makes new data appear in dashboard.

Some factor might be here, that updates are via the query (?) fields - `events` and `paginatedEvents`.

Reply:
I can't seem to reproduce the bug.  Here are the steps I took:

1. go to localhost:3000
2. press `F5`
3. navigate to dashboard/events by clicking Dashboard and then Events
4. click on the edit button of the top event (since that appears on the homepage)
5. change the name
6. click submit
7. see new name in dashboard/events
8. click the freecodecamp logo to go back to localhost:3000/
9. see new name

one thing I noticed was that the dashboard is updating regardless even if you delete the cache update logic. For some reason it's re-querying anyway, but the landing page doesn't update unless you actively modify the cache.

Reply:
Oh, that's interesting.  I should play around with Apollo devtools more!

> Sorry, I might have described it incorrectly.

Nope, not at all. I was just confused that everything *appeared* to be working without the return, but I think I was just getting lucky.

Anyway, thanks for the detailed breakdown - your original approach definitely makes sense.

Reply:
I'm definitely still somewhat confused why it's working this way.